### PR TITLE
Reporter Updates

### PR DESCRIPTION
**Pull request by kdiluca**
- added new environ vars, expiry to setting in cache, use pickle, added check for stale time
- added defaults for env vars

### CODE REVIEW

**Comment by kevinkreiser** [comment]
this looks good to me but im not sure why the test are failing

File: py/reporter_service.py [hunk 1]
```diff
@@ -94,31 +95,46 @@ def parse_trace(self, post):
...
-        #TODO: we will need to prepend the last bit of shape from the partial_end segment that's already in Redis 
-        # to the rest of the partial_start segment once it is returned from the segment_matcher
-        #params['partial_start'] = json.loads(partial_kv)
+      partial = self.server.cache.get(uuid)
+      if partial:
+        partial = pickle.loads(partial)
+        time_diff = trace['trace'][0]['time'] - partial[-1]['time']
+        #check to make sure time is not stale and not in future
+        if time_diff < os.environ.get('STALE_TIME', 60) and time_diff >= 0:
+          #We will need to prepend the last bit of shape from the partial_end segment that's already in Redis
```
Comment (kevinkreiser) [inline py/reporter_service.py]: this comment reads like a `TODO` which has already been done

File: py/reporter_service.py [hunk 2]
```diff
@@ -94,31 +95,46 @@ def parse_trace(self, post):
...
+        if time_diff < os.environ.get('STALE_TIME', 60) and time_diff >= 0:
+          #We will need to prepend the last bit of shape from the partial_end segment that's already in Redis 
+          # to the rest of the partial_start segment once it is returned from the segment_matcher
+          trace['trace'] = partial + trace['trace']
     else:
       return 400, 'No uuid in segment_match request!'
 
     #ask valhalla to give back OSMLR segments along this trace
-    result = self.server.segment_matcher.Match(json.dumps(trace))
+    #temporary - for debugging
```
Comment (kevinkreiser) [inline py/reporter_service.py]: probably wanted to remove this from the pr

**Comment by kdiluca** [comment]
This is currently failing because we need to update the PPA with the shape indices.  For now, we need to work from this branch.

**Pull request merged**